### PR TITLE
Some changes to peer scoring

### LINKED ISSUE #4264: enable peer scoring
need to check how it works etc, not as simple as enabling.

### PR DESCRIPTION
**Pull request by zilm13**
## PR Description
Made some changes with current peer scoring. Not sure dropping incoming banned peers works. Is it a correct place for that?
What I don't like in the current scoring:
- permanence of ban. We don't store it on a disk, but many peers could run for a month without update and ban will stay in place
- lack of transparency in reasons, when applying `unsuitable`. The list of disconnection reasons could be extended with time, we already had ban when we were locally shutting down, not an issue w/o persistence, but could cause an error in development. So I'm totally for ban reasons list instead of ban exclusions.

I've tried to fix it, ran it few times locally, currently testing on set of test nodes

## Documentation

- [x] I thought about documentation and added the `doc-change-required` label to this PR if updates are required.

## Changelog

- [x] I thought about adding a changelog entry, and added one if I deemed necessary.

### CODE REVIEW

**Comment by zilm13** [comment]
About failed tests: looks like we need a way to make static peers protected from any connection restrictions. Though master version was not subject to this issue, I think it's anyway a good improvement to the peer scoring for future extending if static peers are protected. Will fix it.

**Comment by zilm13** [comment]
removed disconnect of incoming banned peer for a while

**Comment by zilm13** [comment]
Ok, I think it's ok to add it in this way, no any incoming bans, but some changes to avoid permanent ban as nodes could run without restart for a weeks.

File: networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPeer.java [hunk 1]
```diff
@@ -89,11 +89,13 @@ public void checkPeerIdentity() {
     getAgentVersionFromIdentity()
         .thenAccept(
             maybeAgent -> {
-              LOG.debug("Connected peer has agent string: {}", maybeAgent.orElse("Unknown"));
+              LOG.debug(
+                  "Connected peer id {} ({}) has agent string: {}",
+                  peerId,
+                  connection.remoteAddress(),
+                  maybeAgent.orElse("Unknown"));
```
Comment (ajsutton) [inline networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPeer.java]: ooo, we should remove this log line or at least the agent string. It's not safe to log arbitrary content from external sources.  We've seen attacks where the peer agent was set to things like "Your Geth is out of date, click <malicious url> to upgrade".

File: networking/p2p/src/test/java/tech/pegasys/teku/networking/p2p/reputation/ReputationManagerTest.java [hunk 2]
```diff
@@ -31,7 +31,7 @@
 class ReputationManagerTest {
```
Comment (ajsutton) [inline networking/p2p/src/test/java/tech/pegasys/teku/networking/p2p/reputation/ReputationManagerTest.java:34]: We should add tests to cover the different short and long bans we're now using.

**Review by ajsutton — COMMENTED** [review]
Looks good generally, just need to make sure we add tests for the new behaviour.

File: networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPeer.java (hunk 1 above)
Comment (zilm13) [reply networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPeer.java]: Maybe make it trace? It was already there, I've just added `peerId` and remote address

Comment (zilm13) [reply networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPeer.java]: ok, 2 votes, removing :)

Comment (zilm13) [reply networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/libp2p/LibP2PPeer.java]: it's useful debug feature to understand what type and version of client is connecting to you

**Comment by zilm13** [comment]
Acceptance looks flaky, not reproduced locally

**Comment by zilm13** [comment]
rerun is ok

**Review by ajsutton — APPROVED** [review]
LGTM.

**Pull request merged**